### PR TITLE
HEL-112 | Showcase model

### PR DESCRIPTION
**Pull request by tuovinensanttu**
Added new model for showing albums on Front page.

### CODE REVIEW

File: hkm/views/views.py [hunk 1]
```diff
@@ -374,19 +374,28 @@ def get_context_data(self, **kwargs):
...
 
     def get_context_data(self, **kwargs):
         context = super(HomeView, self).get_context_data(**kwargs)
-        context['frontpage_images'] = self.collection_qs.filter(
-        show_in_landing_page=True)
-        context['featured_collections'] = self.collection_qs.filter(
-            is_featured=True)
+        context['frontpage_images'] = self.collection_qs.order_by('created')
+        context['showcases'] = self.showcase_qs
         return context
```
Comment (tuomas777) [inline hkm/views/views.py:399]: About the whole view, as it doesn't inherit from `BaseCollectionListView` anymore, IMO there is no point to handle the querysets so complexly, I would do just something like
```
class HomeView(BaseView):
    template_name = 'hkm/views/home_page.html'
    url_name = 'hkm_index'

    def get_context_data(self, **kwargs):
        context = super(HomeView, self).get_context_data(**kwargs)
        context['frontpage_images'] = Collection.objects.filter(show_in_landing_page=True).order_by('created')
        context['showcases'] = Showcase.objects.filter(show_on_home_page=True)
```

Also FWIW the showcases don't have ordering applied, so they will be in random order.

File: hkm/admin.py [hunk 2]
```diff
@@ -59,9 +62,15 @@ class Media:
         js = ('ckeditor/ckeditor.js', 'hkm/js/init.js')
 
 
+
+
```
[the remark this answers is not in the document]
Comment (tuovinensanttu) [reply hkm/admin.py]: Removed.

File: hkm/forms.py [hunk 3]
```diff
@@ -109,4 +110,31 @@ def form_action(self, campaign):
...
+    def __init__(self, *args, **kwargs):
+        super(ShowcaseForm, self).__init__(*args, **kwargs)       
+        collection_ids = []
+        
+        # Find all collections that are owned by museum
+        for collection in Collection.objects.all():
+            user = UserProfile.objects.get(id=collection.owner.id)
+            if user.is_museum and user.user_id == collection.owner.id:
+                collection_ids.append(collection.id)
+        self.fields['albums'].queryset = Collection.objects.all().filter(pk__in=collection_ids)
```
[the remark this answers is not in the document]
Comment (tuovinensanttu) [reply hkm/forms.py]: Thats so much simpler 😅 thanks!

File: hkm/forms.py [hunk 4]
```diff
@@ -109,4 +110,31 @@ def form_action(self, campaign):
...
+
+
+    def clean(self):
+        title = self.cleaned_data.get('title')
```
[the remark this answers is not in the document]
Comment (tuovinensanttu) [reply hkm/forms.py]: Removed.

File: hkm/models/models.py [hunk 5]
```diff
@@ -17,7 +17,7 @@
 from django.core.mail import send_mail
 from django.core.validators import MinValueValidator
 from django.db import models
-from django.db.models.signals import post_save
+from django.db.models.signals import post_save, m2m_changed
```
[the remark this answers is not in the document]
Comment (tuovinensanttu) [reply hkm/models/models.py]: Yea, I forgot to remove this after removing the singal part.

File: hkm/views/views.py (hunk 1 above)
Comment (tuovinensanttu) [reply hkm/views/views.py:385]: That's so much simpler, seems Im little more rusty than I though. I moved `order_by` to showcases, thats where it was supposed to be in the first place.

Comment (tuomas777) [reply hkm/views/views.py:385]: Ok good, but now the collections / frontpage images seem to be in random order? :)

Comment (tuovinensanttu) [reply hkm/views/views.py:385]: Thats ok, i'm taking one of the pictures from that list at random at showing it on the frontpage :)

File: hkm/forms.py [hunk 6]
```diff
@@ -109,4 +110,22 @@ def form_action(self, campaign):
...
+    def __init__(self, *args, **kwargs):
+        super(ShowcaseForm, self).__init__(*args, **kwargs)       
+        self.fields['albums'].queryset = Collection.objects.filter(owner__profile__is_museum=True)
+
+    def clean(self):
+        albums = self.cleaned_data.get('albums')
+
+        if albums:
+            if albums.count() > 3:
+                raise ValidationError(u'Max amount of albums is three')
```
[the remark this answers is not in the document]
Comment (tuovinensanttu) [reply hkm/forms.py]: It should work now, I'll add translations later if somebody wants them.

**Pull request merged**